### PR TITLE
More Makefile improvements

### PR DESCRIPTION
**Pull request by mmokrejs**
Do not override users CFLAGS, CXXFLAGS, CC, CXX variables.

Ensure make stops upon an error in more places.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -57,24 +57,24 @@ all: $(PROGRAM) $(TEST_PROGRAM)
...
 
 # Download and install eigen if not already downloaded
 EIGEN=eigen/INSTALL
 
 $(EIGEN):
-	wget http://bitbucket.org/eigen/eigen/get/3.2.5.tar.bz2
-	tar -xjvf 3.2.5.tar.bz2
-	mv eigen-eigen-bdd17ee3b1b3 eigen
+	if [ ! -e 3.2.5.tar.bz2 ]; then wget http://bitbucket.org/eigen/eigen/get/3.2.5.tar.bz2; fi
+	tar -xjvf 3.2.5.tar.bz2 || ext 255
```
Comment (jts) [inline Makefile]: This should be `exit`

**Comment by jts** [comment]
Thanks for providing this PR. Perhaps rather then using `|| exit 255` it would be cleaner to set the shell to use `pipefail`? https://stackoverflow.com/questions/23079651/equivalent-of-pipefail-in-gnu-make

**Comment by mmokrejs** [comment]
The typo is fixed now. Although it was not intended it was helpfull to see Travis CI was failing as it realized make has failed. So the changes were helpful.

**Comment by mmokrejs** [comment]
> Perhaps rather then using || exit 255 it would be cleaner to set the shell to use pipefail?
I do not know. I had some troubles understanding the syntax how to use the pipefail command at all in the past. So my only comment is:
```
# 'set -o pipefail' turns it on
# 'set +o pipefail' turns it off
```
The concerns about portability of the approach seem valid at the stackoverflow site. But I do not know. Maybe look into some Make generated by say configure from gcc bundle (to see how it propagates back the error codes)?

File: Makefile [hunk 2]
```diff
@@ -9,12 +9,12 @@ SUBDIRS := src src/hmm src/thirdparty src/common src/alignment
...
+CXXFLAGS ?= -g -O3
 CXXFLAGS += -std=c++11 -fopenmp
-CFLAGS=-O3
-CXX=g++
-CC=gcc
-HDF5=install
+CFLAGS ?= -O3
+CXX ?= g++
+CC ?= gcc
+HDF5=install # change to any value to disable compilation of bundled HDF5 code
```
Comment (jts) [inline Makefile]: This inline comment breaks the auto-installation of HDF5, please put the comment on the preceding line

**Pull request merged**